entropy/coin_flip/app/src/flip_coin.ts

```diff
@@ -29,75 +45,80 @@ const parser = yargs(hideBin(process.argv))
 
 async function main() {
   const argv = await parser.argv;
+  const chainName = argv.chainName;
+  if (chainName !== "optimism-sepolia" && chainName !== "arbitrum-sepolia") {
+    throw new Error("Invalid chain name");
+  }
+
+  let chain: Chain;
+  if (chainName === "optimism-sepolia") {
+    chain = optimismSepolia;
+  } else {
+    chain = arbitrumSepolia;
+  }
+
+  const client = createWalletClient({
+    chain: chain,
+    account: privateKeyToAccount(argv.privateKey as Hex),
+    transport: http(argv.rpcUrl),
+  }).extend(publicActions);
+
+  const coinFlipContract = getContract({
+    address: argv.address as Hex,
+    abi: ICoinFlipAbi,
+    client: client,
```

Comment:
Small nit but you can just pass `client`, you don't have to write `client: client`, i.e. `const foo = { bar };` is a shorthand for `const foo = { bar: bar };`